config/webpack.config.js

```diff
@@ -52,11 +65,22 @@ var config = {
       path.join(__dirname, '..', 'webpack'),
       'node_modules/'
     ],
-    alias: {
+    alias: Object.assign({
       foremanReact:
         path.join(__dirname,
-           '../webpack/assets/javascripts/react_app/components')
-    }
+           '../webpack/assets/javascripts/react_app/components'),
+      foremanReactCommon:
+        path.join(__dirname,
+          '../webpack/assets/javascripts/react_app/common'),
+      foremanReactService:
+        path.join(__dirname,
+          '../webpack/assets/javascripts/services'),
+      foremanNodeModules:
```

Comment:
I agree it is a nuisance, but `import React from 'react';` in plugins does not work an I haven't found a better solution yet. I am not sure what you mean about packages that include common other libs, could you explain please?